src/rulesets/Base/mapreduce.jl

```diff
@@ -10,9 +10,10 @@ function rrule(::typeof(sum), x::AbstractArray{T}; dims=:) where {T<:Number}
     y = sum(x; dims=dims)
     function sum_pullback(ȳ)
         # broadcasting the two works out the size no-matter `dims`
-        x̄ = broadcast(x, ȳ) do xi, ȳi
-            ȳi
-        end
+        x̄ = InplaceableThunk(
+            @thunk(broadcast((_,y1)->y1, x, ȳ)), # last∘tuple
```

Comment:
```suggestion
            @thunk(broadcast((_, ȳi) -> ȳi, x, ȳ)),  # last∘tuple
```
BlueStyle spacing.
and calling things same name as before

Reply:
Thoughts on `@thunk(broadcast(last∘tuple, x, ȳ))`? It's a slightly weird broadcast, to just pick one argument, which might be why no very pretty way to write it has evolved.

Reply:
I did think about it.
I am down with it.

Reply:
Let's try this:
```suggestion
            @thunk(broadcast(last∘tuple, x, ȳ))
```